src/main/java/net/minecraftforge/fml/network/NetworkRegistry.java

```diff
@@ -47,6 +51,21 @@
     private static final Marker NETREGISTRY = MarkerManager.getMarker("NETREGISTRY");
 
     private static Map<ResourceLocation, NetworkInstance> instances = new HashMap<>();
+    static final Map<ResourceLocation, Supplier<Function<ByteBuf, GuiScreen>>> guiHandlers = new ConcurrentHashMap<>();
+
+    /**
+     * Registers a client-side GUI handler for the given ID.
+     * The function takes any extra data provided to {@link net.minecraft.entity.player.EntityPlayer#openGui}
+     * and returns a {@link GuiScreen} to display.
+     * Call this during {@link net.minecraftforge.fml.event.lifecycle.FMLClientSetupEvent}.
+     * This method is safe to call in parallel mod loading
+     * @param id
+     * @param handler
+     */
+    public static void registerGui(ResourceLocation id, Supplier<Function<ByteBuf, GuiScreen>> handler)
```

Comment:
I'm not sure if I need the extra Supplier layer here to guard against client-only classes loading. Do I?

Reply:
I don't think you need it. This method should be called only on the client distribution, and the GuiScreen is erased at runtime